grype/db/v5/store/store.go

```diff
@@ -290,7 +290,7 @@ func (s *store) Close() {
 	s.db.Exec("VACUUM;")
 
 	sqlDB, err := s.db.DB()
-	if err != nil {
+	if err == nil {
```

Comment:
Wouldn't we just always want to try to close this, regardless of error status?

Reply:
Indeed, it seems that the DB method might return both an error and a valid instance of the DB. Changed the logic. Thanks!